pkg/sentry/socket/netfilter/netfilter.go

```diff
@@ -466,12 +474,34 @@ func SetEntries(stack *stack.Stack, optVal []byte) *syserr.Error {
 		}
 	}
 
+	// Add the user chains.
+	for ruleIdx, rule := range table.Rules {
+		target, ok := rule.Target.(*iptables.UserChainTarget)
+		if !ok {
+			continue
+		}
+
+		// We found a user chain. Before inserting it into the table,
+		// check that:
+		// - There's some other rule after it.
+		// - There are no matchers.
+		if ruleIdx == len(table.Rules)-1 {
+			log.Warningf("Found a user chain without any rules or default policy.")
```

Comment:
Per above, I don't think Warning are the correct thing to use here. You should think of them like kernel WARN() instances. These are warnings intended for the caller of the API, which is a different audience.

You could make these Infof or Debugf and it would be more reasonable, but Warningf will always be triggered, which is very questionable.

Reply:
(This comment applies broadly in this case, I think that unless the warning indicates some fundamental error, e.g. a programming error such as a recoverable precondition violation, you should probably stick to Infof and Debugf.)